contributed_definitions/NXcs_computer.nxdl.xml

```diff
@@ -57,24 +52,87 @@
-    <group type="NXcs_cpu_sys">
-        <doc>
-             Details about (classical) processing units (CPUs) or a system thereof.
-        </doc>
-    </group>
-    <group type="NXcs_gpu_sys">
+    <group name="processing" type="NXobject">
         <doc>
-             Details about coprocessor/graphic cards/accelerator units or a system thereof.
+             Details about the system of processing units e.g. (classical) processing units (CPUs),
+             coprocessor, graphic cards, accelerator processing units or a system of these.
         </doc>
+        <group type="NXcircuit">
+            <doc>
+                 Granularizing the processing units. Typical examples, a desktop computer
+                 with a single CPU one could describe using one instance of NXcircuit.
+                 A dual-socket server one could describe using two instances NXcircuit
+                 A server with two dual-socket server nodes one could describe with
+                 four instances of NXcircuit surplus a field with their level in the hierarchy.
+            </doc>
+            <field name="type" type="NX_CHAR">
+                <doc>
+                     General type of the processing unit
+                </doc>
+                <enumeration>
+                    <item value="cpu"/>
+                    <item value="gpu"/>
+                    <item value="fpga"/>
+                    <item value="other"/>
+                </enumeration>
+            </field>
+            <field name="name" type="NX_CHAR">
+                <doc>
+                     Given name
+                </doc>
+            </field>
+        </group>
     </group>
-    <group type="NXcs_mm_sys">
+    <group name="memory" type="NXobject">
         <doc>
              Details about the memory sub-system.
         </doc>
+        <group type="NXcircuit">
+            <doc>
+                 Granularizing the components of the memory system.
+            </doc>
+            <field name="technology" type="NX_CHAR">
+                <doc>
+                     Qualifier for the type of random access memory.
+                </doc>
+            </field>
+            <field name="max_physical_capacity" type="NX_NUMBER" units="NX_ANY">
+                <doc>
+                     Total amount of data which the medium can hold.
+                </doc>
+            </field>
+            <field name="name" type="NX_CHAR">
+                <doc>
+                     Given name
+                </doc>
+            </field>
+        </group>
     </group>
-    <group type="NXcs_io_sys">
+    <group name="storage" type="NXobject">
         <doc>
              Details about the I/O sub-system.
         </doc>
+        <group name="ioID" type="NXcircuit">
+            <field name="technology" type="NX_CHAR">
+                <doc>
+                     Qualifier for the type of storage medium used.
+                </doc>
+                <enumeration>
```

Comment:
`Technology does not fit here. What do you think about `storage_type`?

Reply:
Implemented